integration-tests/tests/src/tests/observable.ts

```diff
@@ -490,6 +604,127 @@ describe("Observable", () => {
         },
       });
     });
+
+    describe("key-path filtered", () => {
+      it("calls listener only on relevant changes", async function (this: RealmObjectContext<Person>) {
+        const collection = this.realm.objects<Person>("Person").filtered("name = $0 OR age = 42", "Alice");
+        await expectCollectionNotifications(
```

Comment:
Perhaps named params would be a bit clearer.  I find myself having to jump back and forth to the implementation to follow what all these mean.

Reply:
Something like:
```
expectCollectionNotifications(
collection,
[
 {result: EMPTY_COLLECTION_CHANGESET},
 {action: () => /*..*/, result: {deletions:[], /*etc*/ }}
 //...
],
{
 keypaths: ["name"]
}
)
```

Reply:
Perhaps if we are expecting this to initialize with `EMPTY_COLLECTION_CHANGESET`, then we could just assert this within the utility, as it doesn't provide much value into the test itself to know this is being asserted.

Reply:
The `expectXNotifications` functions are already pretty magical, I fear including this in those will make them even harder to reason about as they'll start making assumptions on the behaviour of the SDK too: I like these empty changesets are explicitly listed (even if its more verbose).

Reply:
>  I find myself having to jump back and forth to the implementation to follow what all these mean.

Between the implementation of the `expectXNotifications` and the tests? Isn't it pretty clear once you've understand that they just assert changeset state or execute the functions? Having the `action` as part of an object with `result`, I fear someone could read it as the "action" that needs to be taken after a the `result` has been asserted. In other words, is this a pre-action or a post-action? Technically the `action` needs to fire as part of the function that asserts the previous `result` (if that makes sense).

Reply:
Most likely since I'm in the context of Github PR view, it's a little less intuitive to discern what the intention of the parameters to the `expectNotifications` functions are and how they are executed.  
>  is this a pre-action or a post-action? 

I found myself questioning that in its current form. They are quite magical, so I would be happy if we provided guidance through named parameters or through documentation.

Reply:
I updated the documentation of a shared utility function. I played around with named parameters, but couldn't find short names that really captured the intent without feeling ambiguous to me. I think having a list of "changes and actions" which are either asserted or executed in an ordered sequence is pretty easy to reason about, with that we don't need to explain if the `action` goes before or after the change / `result` in the same object.